AnnService/src/Core/VectorIndex.cpp

```diff
@@ -114,8 +114,10 @@ VectorIndex::BuildMetaMapping()
 {
     m_pMetaToVec.reset(new std::unordered_map<std::string, SizeType>);
     for (SizeType i = 0; i < m_pMetadata->Count(); i++) {
-        ByteArray meta = m_pMetadata->GetMetadata(i);
-        m_pMetaToVec->emplace(std::string((char*)meta.Data(), meta.Length()), i);
+        if (ContainSample(i)) {
+            ByteArray meta = m_pMetadata->GetMetadata(i);
+            m_pMetaToVec->emplace(std::string((char*)meta.Data(), meta.Length()), i);
```

Comment:
if i not contains in m_pMetadata expected? should we add some error handle code?

Reply:
Here we check whether this vector has been deleted or not. If we do not check this, the metadata will be overwritten by deleted items.